### PR TITLE
Add total_tft and total_unlocked_tft Endpoints

### PR DESCRIPTION
**Pull request by MElborolossy**
### Description

Add total_tft and total_unlocked_tft endpoints for stellar_stats

### Related Issues
- Fix threefoldfoundation/tft-stellar#206

### Checklist

- [x] Build pass
- [x] Documentation
- [x] Code format and docstrings

### CODE REVIEW

**Review by xmonader — APPROVED** [review]

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 1]
```diff
@@ -49,4 +49,32 @@ def get_stats():
...
 
+@app.route("/api/total_tft")
+def total_tft():
+    query_params = request.query.decode()
+    network = query_params.get("network", "public")
+    tokencode = query_params.get("tokencode", "TFT")
+    detailed = j.data.serializers.json.loads(query_params.get("detailed", "false"))
+
+    collector = StatisticsCollector(network)
+    stats = collector.getstatistics(tokencode, detailed)
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:60]: this process takes a very long time, maybe we can introduce caching for 10 mins using redis like the get_stats method. if the results are there in redis fine get it directly if no, we can set it

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 2]
```diff
@@ -49,4 +49,32 @@ def get_stats():
...
+
+    return f"{stats['total']:,.7f}"
+
+
+@app.route("/api/total_unlocked_tft")
+def total_tft():
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:66]:
```suggestion
def total_unlocked_tft():
```

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 3]
```diff
@@ -49,4 +49,32 @@ def get_stats():
...
+    query_params = request.query.decode()
+    network = query_params.get("network", "public")
+    tokencode = query_params.get("tokencode", "TFT")
+    detailed = j.data.serializers.json.loads(query_params.get("detailed", "false"))
+
+    collector = StatisticsCollector(network)
+    stats = collector.getstatistics(tokencode, detailed)
+    total_tft = stats["total"]
+    total_locked_tft = stats["total_locked"]
+    total_unlocked_tft = total_tft - total_locked_tft
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:76]: also here we can cache the results

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 4]
```diff
@@ -49,4 +49,32 @@ def get_stats():
...
+    return f"{total_unlocked_tft:,.7f}"
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:77]: the requested result should be float

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 5]
```diff
@@ -49,4 +49,32 @@ def get_stats():
...
+def total_tft():
+    query_params = request.query.decode()
+    network = query_params.get("network", "public")
+    tokencode = query_params.get("tokencode", "TFT")
+    detailed = j.data.serializers.json.loads(query_params.get("detailed", "false"))
+
+    collector = StatisticsCollector(network)
+    stats = collector.getstatistics(tokencode, detailed)
+
+    return f"{stats['total']:,.7f}"
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:62]: the requested result should be float

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 6]
```diff
@@ -49,4 +49,49 @@ def get_stats():
...
+    cached_data = redis.get(f"{network}-{tokencode}-total_tft")
+    if cached_data:
+        return cached_data
+
+    collector = StatisticsCollector(network)
+    stats = collector.getstatistics(tokencode, False)
+
+    total = stats["total"]
+    redis.set(f"{network}-{tokencode}-total_tft", total, ex=600)
+    return f"{total}"
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:69]: we want to return a float value to the user as described in the issue

```suggestion
    return total
```

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 7]
```diff
@@ -49,4 +49,49 @@ def get_stats():
...
+    # return f"{stats['total']:,.7f}"
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:70]:
```suggestion
    
```

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 8]
```diff
@@ -49,4 +49,49 @@ def get_stats():
...
+
+    collector = StatisticsCollector(network)
+    stats = collector.getstatistics(tokencode, False)
+
+    total_tft = stats["total"]
+    total_locked_tft = stats["total_locked"]
+    total_unlocked_tft = total_tft - total_locked_tft
+
+    redis.set(f"{network}-{tokencode}-total_unlocked_tft", total_unlocked_tft, ex=600)
+    return f"{total_unlocked_tft}"
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:93]: we want to return a float value to the user as described in the issue

```suggestion
    return total_unlocked_tft
```

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py [hunk 9]
```diff
@@ -49,4 +49,49 @@ def get_stats():
...
+    # return f"{total_unlocked_tft:,.7f}"
```
Comment (waleedhammam) [inline jumpscale/packages/stellar_stats/bottle/stellar_stats.py:94]:
```suggestion
    
```

File: jumpscale/packages/stellar_stats/bottle/stellar_stats.py (hunk 6 above)
Comment (MElborolossy) [reply jumpscale/packages/stellar_stats/bottle/stellar_stats.py:69]: This cause 500 error

**Review by waleedhammam — APPROVED** [review]

**Pull request merged**